### PR TITLE
Manufacturer page fixes

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Changed manufacturer address grid name field null values to display as '--'.
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | In `Sell > Catalog > Brands & Suppliers` if brand is not selected in address grid, `'--'` should appear in field name instead of previously shown empty field

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13164)

### CODE REVIEW

**Comment by zuk3975** [comment]
@matks ready.  #9611, #9610 seems to be handled already in brands migration

**Review by matks — APPROVED** [review]

File: src/Core/Grid/Data/Factory/ManufacturerAddressGridDataFactory.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            $addresses->getQuery()
+        );
+    }
+
+    /**
+     * @param array $addresses
+     *
+     * @return array
+     */
+    private function applyModification(array $addresses)
```
Comment (mickaelandrieu) [inline src/Core/Grid/Data/Factory/ManufacturerAddressGridDataFactory.php:74]: wtf? why are we doing private changes here so the PrestaShop developers won't be able to change or alter them?

Comment (mickaelandrieu) [reply src/Core/Grid/Data/Factory/ManufacturerAddressGridDataFactory.php:74]: needs to be configurable in the next minors versions (like the applicators introduced by @eternoendless :+1: )

**Review by mickaelandrieu — APPROVED** [review]

File: src/Core/Grid/Data/Factory/ManufacturerAddressGridDataFactory.php (hunk 1 above)
Comment (sarjon) [reply src/Core/Grid/Data/Factory/ManufacturerAddressGridDataFactory.php:74]: I think developers can already do it (via hooks), but we can't, that's why we need a decorator.

Comment (matks) [reply src/Core/Grid/Data/Factory/ManufacturerAddressGridDataFactory.php:74]: @mickaelandrieu I dont understand the issue o_O. If a developer wants to alter this behavior, he can:
- override the Symfony service with another service
- use another decorator to add his behavior

Did you want the function to be `protected` ? `public` ? Then ` getData(SearchCriteriaInterface $searchCriteria)` is already public, why do we need `applyModification()` to be opened as well ?

Comment (mickaelandrieu) [reply src/Core/Grid/Data/Factory/ManufacturerAddressGridDataFactory.php:74]: > override the Symfony service with another service

it should be done via a hook, remember that Symfony provides an extra experience, not a replacement (in 1.7).

> use another decorator to add his behavior

what do you mean? to be able to decorate the modifications applied to the data you should be able to have the hand on the " data modifiers" using a hook: this is what I'm suggesting here. 

> I think developers can already do it (via hooks)

I may be wrong but the current hook is called after this function call, so the developers won't have access to the unaltered data, isn't it?

Comment (sarjon) [reply src/Core/Grid/Data/Factory/ManufacturerAddressGridDataFactory.php:74]: > I may be wrong but the current hook is called after this function call, so the developers won't have access to the unaltered data, isn't it?

Yep, I think you're right.

Some time ago I wanted to introduce `Modifiers`, that could be hooked in (by registering a service & tagging it) for any Grid by either module developers or core developer.

Comment (PierreRambaud) [reply src/Core/Grid/Data/Factory/ManufacturerAddressGridDataFactory.php:74]: > I may be wrong but the current hook is called after this function call, so the developers won't have access to the unaltered data, isn't it?

Weird.. We must be able to alter data when it hasn't been altered :thinking: 
Should be important to add a hook here.

**Comment by matks** [comment]
Restarting Travis build as it seems we never got result report 🤔

**Comment by matks** [comment]
Thank you @zuk3975 

@mickaelandrieu, @sarjon : debate is not closed, we need to discuss this further 😉

**Pull request merged**